### PR TITLE
Efficient automated GitHub testing

### PR DESCRIPTION
**Pull request by kathrynle20**
Optimized automated GitHub testing to run faster with these changes:
- Created additional value for configs called "workflow_test". Set this value to True if you want to run in testing mode
- Testing mode will reduce the test set size. Currently the test set size is reduced from 10,000 to 1,000.
- Made changes to main_grpc.py to include an extra argument to indicate that we want to spin nodes that run in testing mode. Set the argument -dev to True if testing and False if not.

Result: Test Training Code with gRPC GitHub Action runs in 4 mins compared to 13 mins previously.

### CODE REVIEW

**Comment by kathrynle20** [comment]
Testing now runs on the fedstatic algorithm. The fedstatic server now downloads the dataset.

File: src/configs/sys_config.py [hunk 1]
```diff
@@ -158,10 +158,10 @@ def get_digit_five_support(num_users: int, domains: List[str] = DIGIT_FIVE):
 CIAR10_DPATH = "./datasets/imgs/cifar10/"
 
 NUM_COLLABORATORS = 1
-# DUMP_DIR = "../../../../../../../home/"
-DUMP_DIR = "/tmp/"
+DUMP_DIR = "/Users/kathryn/MIT/UROP/Media Lab/sonar_experiments/"
```
Comment (tremblerz) [inline src/configs/sys_config.py:161]: Can you change this.

**Review by tremblerz — CHANGES_REQUESTED** [review]
